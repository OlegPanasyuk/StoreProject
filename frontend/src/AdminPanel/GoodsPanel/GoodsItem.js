import React, { Component } from 'react';
import { Card, Button, Col, Row, ButtonGroup } from 'react-bootstrap';
import PropTypes from 'prop-types';

//Redux 
import { connect } from 'react-redux';
import {
    openEditGoodsItem,
    permissionToDelete
} from '../../REDUX/adminPanel/actions/actionsGoodsPanel';

export class GoodsItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            id_img: 0
        };
        this.id_img = 0;
    }

    UNSAFE_componentWillMount() {
        let self = this;
        if (fetch) {
            let myInit = {
                method: 'GET',
                cache: 'default'
            };

            fetch(`${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/images/goods/${this.props.obj.idgoods}`, myInit)
                .then(res => {
                    if (res.status === 200) {
                        return res.json();
                    }
                })
                .then(img => {
                    self.setState({
                        id_img: img[0].imgs_id_img
                    });

                });

        }
    }

    render() {
        let { obj } = this.props;
        let { id_img } = this.state;
        let src = '';

        src = `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/images/${id_img}`;

        return (
            <Card className='mb-3'>
                <Card.Header >

                    <Row>
                        <Col>
                            <Card.Img variant="top" src={src} style={{
                                width: '50%'
                            }} />
                            <Card.Title>
                                {obj.name}
                            </Card.Title>
                            <Card.Text>
                                {`${obj.price} $`}
                            </Card.Text>
                        </Col>
                        <ButtonGroup
                            className='mr-3'
                        >
                            <Button
                                variant='light'
                                onClick={() => this.props.openEditGoodsItem({
                                    ...obj,
                                    id_img,
                                    show: true
                                })}
                            >
                                <i className="far fa-edit"></i>
                            </Button>
                            <Button
                                variant='light'
                                onClick={() => {
                                    this.props.permissionToDelete(obj.idgoods);
                                }}
                            >
                                <i className="fas fa-trash-alt"></i>
                            </Button>
                        </ButtonGroup>
                    </Row>

                </Card.Header>
                <Card.Body className='d-flex flex-column'>
                    <Card.Text style={{ padding: '10px' }}>
                        {obj.description}
                    </Card.Text>

                </Card.Body>
            </Card>
        );
    }
}

GoodsItem.propTypes = {
    obj: PropTypes.object,
    id: PropTypes.number,
    openEditGoodsItem: PropTypes.func,
    permissionToDelete: PropTypes.func
};


export default connect(null, {
    openEditGoodsItem,
    permissionToDelete
})(GoodsItem);