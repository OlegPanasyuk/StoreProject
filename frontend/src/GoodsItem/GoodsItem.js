import React, { Component } from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';

class GoodsItem extends Component {
    render() {
        let { obj } = this.props;
        return (
            <Card>
                <Card.Header>
                    {obj.name}
                </Card.Header>
                <Card.Text style={{ padding: '10px' }}>
                    {obj.description}
                </Card.Text>
            </Card>
        );
    }
}

GoodsItem.propTypes = {
    obj : PropTypes.object
};


export default GoodsItem;