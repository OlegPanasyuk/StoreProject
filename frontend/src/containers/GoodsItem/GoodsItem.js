import React, { Component } from 'react';
import { Card, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addGoodsToBasket } from '../../actions/actionsShoppingBasket';
import imgDef from '../../asserts/imgs/header.jpg';
import ImageService from '../../Services/ImageService';

export class GoodsItem extends Component {
    state = {
        url: ''
    }

    handleAddToBasket = () => {
        const { addGoodsToBasket, id } = this.props;
        addGoodsToBasket(id);
    };

    // eslint-disable-next-line camelcase
    UNSAFE_componentWillMount() {
        const { obj } = this.props;
        ImageService.getGoodsImage({
            idGoods: obj.idgoods
        }).then((res) => {
            const { data } = res;
            if (data) {
                data.forEach((el) => {
                    if (el.title > 0) {
                        this.setState({
                            url: `${
                                process.env.REACT_APP_API_HOST
                            }:${
                                process.env.REACT_APP_API_PORT
                            }/images/${el.imgs_id_img}`
                        });
                    }
                });
            }
        });
    }


    render() {
        const { obj } = this.props;
        let { url } = this.state;
        if (url.length === 0) {
            url = imgDef;
        }
        return (
            <Card>
                <div style={{
                    position: 'relative'
                }}
                >
                    <Card.Img variant='top' src={url}>

                    </Card.Img>
                    <div
                        className='card-img-overlay d-flex flex-column justify-content-end'
                        style={{
                            color: 'var(--white)',
                            background: 'linear-gradient(to top, var(--dark) 5%, transparent 60%)'
                        }}
                    >
                        <Card.Title>
                            {obj.name}
                        </Card.Title>
                        <Card.Text>
                            {`${obj.price} $`}
                        </Card.Text>
                    </div>
                </div>


                <Card.Body className='d-flex flex-column'>
                    <Card.Text style={{}}>
                        {obj.description}
                    </Card.Text>
                    <Button onClick={this.handleAddToBasket}>
                        Add to backet
                    </Button>
                </Card.Body>
            </Card>
        );
    }
}

GoodsItem.propTypes = {
    obj: PropTypes.object,
    id: PropTypes.number,
    addGoodsToBasket: PropTypes.func
};

GoodsItem.defaultProps = {
    obj: {},
    id: 0,
    addGoodsToBasket: () => {}
};

export default connect(null, { addGoodsToBasket })(GoodsItem);
