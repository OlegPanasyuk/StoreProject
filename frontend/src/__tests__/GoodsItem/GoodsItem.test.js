import React from 'react';
import GoodsItem from '../../GoodsItem/GoodsItem';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import rest from 'rest';

configure({ adapter: new Adapter() });

import pathPrefix from 'rest/interceptor/pathPrefix';
import errorCode from 'rest/interceptor/errorCode';
import mime from 'rest/interceptor/mime';

const client = rest.wrap(mime, { mime: 'application/json' })
    .wrap(errorCode, { code: 500 })
    .wrap(pathPrefix, { prefix: 'http://localhost:3300' });


function addItemGoodsToDB(obj = {
    name: 'TestFromTests',
    description: 'Lorem Test For Tests and Only Tests',
    catalogue_id_catalogue: 5
}) {
    return new client({ method: 'POST', path: 'goods', entity: obj });
}

function deleteItemGoodsFromDB(id) {
    let query = '?id=' + id;
    client({ method: 'DELETE', path: 'goods' + query }).then(() => {

    });
}
function setupComponent(props = {
    obj: {
        idgoods: 1,
        name: 'Honda Civic',
        description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. \
        Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates \
        quis voluptate, dolorem modi non? Nisi, maxime a.',
        catalogue_id_catalogue: 4
    }
}) {
    let component = shallow(<GoodsItem {...props} />);
    return { component, props };
}

// eslint-disable-next-line no-undef
describe('GoodsItem ', () => {
    // eslint-disable-next-line no-undef
    it('Component must be created', () => {
        const { component } = setupComponent();

        // eslint-disable-next-line no-undef
        expect(GoodsItem.propTypes.obj).toBe(PropTypes.object);
        // eslint-disable-next-line no-undef
        expect(shallowToJson(component)).toMatchSnapshot();

    });
    // eslint-disable-next-line no-undef
    it('Component must be rendered after creation', () => {
        let obj = {};

        return addItemGoodsToDB()
            .then((res) => {
                obj = res.entity;
                const { component } = setupComponent({ obj: obj });
                deleteItemGoodsFromDB(obj.idgoods);
                // eslint-disable-next-line no-undef
                expect(shallowToJson(component)).toMatchSnapshot();
            })
            .catch((e) => {
                throw new Error(e);
            });

    });
});