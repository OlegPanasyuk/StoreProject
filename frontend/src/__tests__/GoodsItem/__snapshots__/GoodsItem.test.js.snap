// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoodsItem  Component must be created 1`] = `
<Card
  as="div"
  body={false}
>
  <CardHeader>
    Honda Civic
  </CardHeader>
  <CardText
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit.         Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates         quis voluptate, dolorem modi non? Nisi, maxime a.
  </CardText>
</Card>
`;

exports[`GoodsItem  Component must be rendered after creation 1`] = `
<Card
  as="div"
  body={false}
>
  <CardHeader>
    TestFromTests
  </CardHeader>
  <CardText
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    Lorem Test For Tests and Only Tests
  </CardText>
</Card>
`;
